amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
@tvansteenburgh That doesn't satisfy my requirement of "let me test on different architectures" if the author explicitly puts an arch constraint